src/library_wasmfs.js

```diff
@@ -122,6 +141,22 @@ mergeInto(LibraryManager.library, {
     // TODO: mkdirTree
     // TDOO: rmdir
     // TODO: open
+    open: (path, flags, mode) => {
+      // According to the existing File System docs, Emscripten does not
+      // support user permissions
+
+      // "The underlying implementation does not support user 
+      // or group permissions. The file permissions set in mode 
+      // are only used if the file is created. The caller is always 
+      // treated as the owner of the file, and only those permissions apply."
+      
```

Comment:
```suggestion   
```
Quoting the docs here probably isn't necessary, and runs the risk of going stale.

Reply:
I really appreciate your code review. Thank you. I have answered your questions below. 
1. I remember I originally created test_open_wasmfs.out because of a "file not found" error when I was running a different test. It looks like the method `self.do_runf()` takes in an expected output that it checks on stdout without any need to create out files.
https://github.com/emscripten-core/emscripten/blob/496a4f44d36197601b185d486fb2fb1a5fa19431/test/common.py#L1280-L1281

I think the .out file is necessary for running tests with `self.do_run_in_out_file_test"`, so I deleted the `.out` file and the test still passed. I'm calling `test_fs_open_wasmfs` with `self.do_runf()` (The one above). 
https://github.com/emscripten-core/emscripten/blob/496a4f44d36197601b185d486fb2fb1a5fa19431/test/common.py#L1287-L1298

2. I did find some other test files that have `FS.open()`: `test_llseek.c`, `test_nodefs.c`, and `test_write.cpp`. All three tests call some fs functions that aren't yet in `library_wasmfs.js` yet so the tests all failed. 

Note - I noticed that my pull request code fails the `test-wasm64` in circleci due to 
```
TypeError: Cannot convert 65488 to a BigInt
    at /tmp/emtest_89oyqghg/emscripten_test_wasm64l_4uf_hulx/test_open_wasmfs.js:909:22
```
From what I'm understading, this seems to be an error with the way I wrote my test, rather than the wasmfs library. However, is it possible that my actual wasmfs code has some issues?

Reply:
Thanks, adding a new test file sounds good to me, then.

I'm not sure off the top of my head what could be going wrong with the wasm64 test. It would be good to dig into that more. Can you at least reproduce the problem locally?